curriculum/challenges/english/02-javascript-algorithms-and-data-structures/debugging/understanding-the-differences-between-the-freecodecamp-and-browser-console.english.md

```diff
@@ -7,31 +7,28 @@ forumTopicId: 301193
-Values passed to <code>console.log()</code> in the text editor block run each set of tests as well as one more time for any function calls that you have in your code.
-This lends itself to some interesting behavior and might trip you up in the beginning, because a logged value that you expect to see only once may print out many more times depending on the number of tests and the values being passed to those tests.
-If you would like to see only your single output and not have to worry about running through the test cycles, you can use <code>console.clear()</code>.
+You may have noticed that some freeCodeCamp JavaScript challenges include their own console. This console behaves a little differently than the browser console you used in the last challenge.
+The following challenge is meant to highlight the main difference between the freeCodeCamp console and your browser console.
+When you run ordinary JavaScript, the browsers console will display your <code>console.log()</code> statements the exact number of times you requested.
+The freeCodeCamp console will print your <code>console.log()</code> statements for each test of that challenge, as well as one more time for any function calls that you have in your code.
+This lends itself to some interesting behavior and might trip you up in the beginning, because a logged value that you expect to see only once may print out many more times.
+If you would like to see only your single output and not have to worry about running through the test cycles, you can use <code>console.clear()</code> and check the browsers console.
 </section>
 
 ## Instructions
 <section id='instructions'>
-Use <code>console.log()</code> to print the variables in the code where indicated.
-
+First, use <code>console.clear()</code> to clear the browser console. After that, use <code>console.log</code> to log the <code>output</code> variable.
 </section>
 
 ## Tests
 <section id='tests'>
 
 ```yml
 tests:
-  - text: Use <code>console.log()</code> to print the <code>outputTwo</code> variable.  In your Browser Console this should print out the value of the variable two times.
-    testString: assert(code.match(/console\.log\(outputTwo\)/g));
-  - text: Use <code>console.log()</code> to print the <code>outputOne</code> variable.
-    testString: assert(code.match(/console\.log\(outputOne\)/g));
-  - text: Use <code>console.clear()</code> to modify your output so that <code>outputOne</code> variable only outputs once.
+  - text: Use <code>console.clear()</code> to clear the browser console.
     testString: assert(code.match(/^(\s*console.clear\(\);?\s*)$/gm));
+  - text: Use <code>console.log()</code> to print the <code>output</code> variable.
+    testString: assert(code.match(/console\.log\(output\)/g));
```

Comment:
To allow for white space, can we use a helper?
```suggestion
    testString: const noSpaces = code.replace(/\s/g, ''); assert(code.match(/console\.log\(output\)/));
```

Reply:
This should work here since we are logging variables. Don't you want it to be like this though...
```suggestion
assert(noSpaces.match(/console\.log\(output\)/));
```
`noSpaces.match` instead of `code.match` like the other suggestion?

Reply:
Yeah, I meant to replace the `code` variable with `noSpaces`.  I will update my suggestion so you can commit it.